infrastructure/demo/main/db.tf

```diff
@@ -0,0 +1,30 @@
+data "aws_rds_engine_version" "postgres" {
+  # The latest version of Postgres is used for the RDS instance.
+  engine             = "postgres"
+  preferred_versions = ["15.3"]
+}
+
+resource "aws_db_instance" "web_db" {
+  # The RDS instance is created.
+  # The RDS instance is created in the private subnet.
+  # The RDS instance is not publicly accessible.
+  # The RDS instance is encrypted.
+  identifier                 = var.rds_db_instance_identifier
+  allocated_storage          = 60
+  port                       = 5432
+  instance_class             = var.rds_db_instance_class
+  storage_type               = "gp2"
+  backup_retention_period    = 7
+  db_name                    = var.rds_db_name
```

Comment:
I get a Terraform error on this line (red underline) like the following:
```
Unexpected attribute: An attribute named "db_name" is not expected here
```
Also, `terraform apply` is hanging at creating this resource -- could this be the reason why?

e.g.
```
module.main.aws_db_instance.web_db: Still creating... [5m30s elapsed]
```

Reply:
This is fine. Dbs take time to build and destroy. [This](https://developer.hashicorp.com/terraform/tutorials/aws/aws-rds) is a reference from there documentation

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance?product_intent=terraform and I see it's an optional parameter also!